settings/src/utilities.js

```diff
@@ -12,6 +12,11 @@ export function getUserRecord( userId ) {
 		userRecord.isSaving = useSelect( ( select ) => select( coreDataStore ).isSavingEntityRecord( 'root', 'user', userId ) );
 	}
 
+	// Initialize the password as an empty string, necessary for resetting incomplete state when leaving the password setting page.
+	if ( userRecord.record && undefined === userRecord.record.password ) {
```

Comment:
@ryelle @adamwoodnz Have we preferred optional chaining in our code bases?

Reply:
I've tried optional chaining, but it didn't work well in this case.

When `useEntityRecord` is called for the first time, `userRecord.record` is undefined. This causes the code to enter the if condition directly and results in an error: `Cannot read properties of undefined (reading 'password')`

We could probably add something like the below to this [Line](https://github.com/WordPress/wporg-two-factor/pull/136/files/fa879b64b45171f2e3c5c9843bc218dc0c393109#diff-d2b2e322863e6dc961ea2ef9b6f6b6e2b5439e53ceb4144f5805b1dd61411b16R9) to fix it.
```
if ( ! userRecord.hasResolved ) {
  return;
}
```